extensions/amp-video/0.1/amp-video.js

```diff
@@ -364,6 +365,49 @@ class AmpVideo extends AMP.BaseElement {
     return promise;
   }
 
+  /**
+   * Gracefully handle media errors if possible.
+   */
+  handleMediaError_() {
+    if (
+      !this.video_.error ||
+      this.video_.error.code != MediaError.MEDIA_ERR_DECODE
+    ) {
+      return;
+    }
+    // HTMLMediaElements automatically fallback to the next source if a load fails
+    // but they don't try the next source upon a decode error.
+    // This code does this fallback manually.
+    user().error(
+      TAG,
+      `Decode error in ${this.video_.currentSrc}`,
+      this.element
+    );
+    // Find the source element that caused the decode error.
+    let sourceCount = 0;
+    const currentSource = childElement(this.video_, (source) => {
+      if (source.tagName != 'SOURCE') {
+        return false;
+      }
+      sourceCount++;
+      return source.src == this.video_.currentSrc;
```

Comment:
Does this work if `src` is a relative URL? I think `currentSrc` will always return the resolved URL. There's a helper in the URL service to resolve the URL.

Other ideas to find the source that caused the error:
- double check if the error `event.target` gives you the source that errored, but unfortunately I don't think it does
- add one error listener per source

Reply:
Yep, if you read the property instead of the attribute it gets expanded. That is also why I didn't use querySelector.

I updated the test to make sure there is coverage for this case.